op-e2e/actions/kromampt_fork_test.go

```diff
@@ -0,0 +1,182 @@
+package actions
+
+import (
+	"context"
+	oppredeploys "github.com/kroma-network/kroma/op-bindings/predeploys"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/kroma-network/kroma/kroma-bindings/bindings"
+	"github.com/kroma-network/kroma/kroma-bindings/predeploys"
+	"github.com/kroma-network/kroma/kroma-chain-ops/genesis"
+)
+
+var (
+	l1BlockMPTCodeHash           = common.HexToHash("0x18fc22e60c9fc3d6a26e9b5c3e4c4e333f7646157e0d59cc18f4da0d43b84227")
```

Comment:
What are the sources of these codehashes?

Reply:
We can get them by launching a devnet, get the each implementation addresses by calling proxy with `implementation()`, and requesting `eth_getProof` to the address. The result of getProof includes codeHash of that contract.